system-addon/content-src/lib/snippets.js

```diff
@@ -342,7 +343,8 @@ function addSnippetsSubscriber(store) {
       initializing = true;
       await snippets.init({appData: state.Snippets});
       initializing = false;
-    } else if (state.Prefs.values["feeds.snippets"] === false && snippets.initialized) {
+    } else if ((state.Prefs.values["feeds.snippets"] === false ||
+      state.Prefs.values.showSnippets === false) && snippets.initialized) {
```

Comment:
I'm assuming the `=== false` is to skip the case where `state.Prefs.initialized` is `false`, so we can make that explicit instead of implied with the first `if` condition. Also, we actually are trying to do the opposite / `else` but it's hidden in the extra conditionals. So I think we can clean it up and be more explicit and hopefully avoid bugs if someone touched one condition but not the other:
```js
// Can't decide if we don't have prefs
if (!state.Prefs.initialized) {
  return;
}
// We should show snippets
if (state.Prefs.values["feeds.snippets"] && state.Prefs.values.showSnippets) {
  if (state.Snippets.initialized && !snippets.initialized && !initializing) {
    initializing = true; …
  }
} else {
  // Shouldn't show snippets
  if (snippets.initialized) {
    snippets.uninit();
  }
}
```

I suppose alternatively, it could be just as clear with a variable, so we can keep it more like what it's now:
```js
// Can't decide if we don't have prefs
if (!state.Prefs.initialized) {
  return;
}
const shouldShow = state.Prefs.values["feeds.snippets"] && state.Prefs.values.showSnippets;
if (shouldShow && …) {
} else if (!shouldShow && …) {
}
```